src/chimera/fix/interpolate/camera.cpp

```diff
@@ -91,8 +91,14 @@ namespace Chimera {
             if(player) {
                 auto *object = ObjectTable::get_object_table().get_dynamic_object(player->object_id);
                 if(object) {
-                    //CHeck if dead and if not, don't interpolate debug/death camera.
-                    if (type == CameraType::CAMERA_DEBUG && object->health >= 0.0) {
+                    // CHeck if dead and if not, don't interpolate debug/death camera.
+                    if(type == CameraType::CAMERA_DEBUG && object->health >= 0.0) {
+                        skip = true;
+                        return;
+                    }
+
+                    // Is the camera moving fast but the player biped velocity low? Probably teleporting so shouldn't interpolate camera.
+                    if(type == CameraType::CAMERA_FIRST_PERSON && distance_squared(previous_tick->data.position, current_tick->data.position) > 0.5 * 0.5 && magnitude(object->velocity) <= 0.5) {
```

Comment:
Since you're comparing against a constant here, you can drop the square root part of the calculation for speed (see the `skip = ` calculation above using `distance_squared`).

Would look something like:
```cpp
magnitude_squared(object->velocity) <= 0.25
```

Reply:
Sure thing.